_modules/elife.py

```diff
@@ -62,6 +63,13 @@ def image_tag():
         return 'latest' # conventionally stable Docker tag
     return revision
 
+def image_label(image, label):
+    """used to retrieve a `label` from an `image`.
+
+    Warning: it will pull the image first, using bandwidth, time and disk space."""
```

Comment:
This is a bit weird, but since we want to place the value of this label in a file, I couldn't find another way to execute some code before populating a `file.managed` template; it has to end up in a new environment variable here: https://github.com/elifesciences/annotations-formula/blob/master/salt/annotations/config/home-deployuser-annotations-.env

Fortunately, the image will be pulled anyway when it's actually used by the following states; so this is not adding time to the build as much as moving the operation up here instead (before any states are even executed I think, which is the misleading part).